pkg/scheduler/scheduler.go

```diff
@@ -97,46 +97,88 @@ type Configurator interface {
+	ecache     *core.EquivalenceCache
+	nodeLister algorithm.NodeLister
+	algorithm  algorithm.ScheduleAlgorithm
+	binder     Binder
+	// podConditionUpdater is used only in case of scheduling errors. If we succeed
 	// with scheduling, PodScheduled condition will be updated in apiserver in /bind
 	// handler so that binding and setting PodCondition it is atomic.
-	PodConditionUpdater PodConditionUpdater
-	// PodPreemptor is used to evict pods and update pod annotations.
-	PodPreemptor PodPreemptor
+	podConditionUpdater PodConditionUpdater
+	// podPreemptor is used to evict pods and update pod annotations.
+	podPreemptor PodPreemptor
 
-	// NextPod should be a function that blocks until the next pod
+	// nextPod should be a function that blocks until the next pod
 	// is available. We don't use a channel for this, because scheduling
 	// a pod may take some amount of time and we don't want pods to get
 	// stale while they sit in a channel.
-	NextPod func() *v1.Pod
+	nextPod func() *v1.Pod
 
-	// WaitForCacheSync waits for scheduler cache to populate.
+	// waitForCacheSync waits for scheduler cache to populate.
 	// It returns true if it was successful, false if the controller should shutdown.
-	WaitForCacheSync func() bool
+	waitForCacheSync func() bool
 
-	// Error is called if there is an error. It is passed the pod in
+	// error is called if there is an error. It is passed the pod in
 	// question, and the error
-	Error func(*v1.Pod, error)
+	error func(*v1.Pod, error)
 
 	// Recorder is the EventRecorder to use
 	Recorder record.EventRecorder
 
 	// Close this to shut down the scheduler.
 	StopEverything chan struct{}
 
-	// VolumeBinder handles PVC/PV binding for the pod.
-	VolumeBinder *volumebinder.VolumeBinder
+	// volumeBinder handles PVC/PV binding for the pod.
+	volumeBinder *volumebinder.VolumeBinder
+}
+
+// NewConfig creates a new scheduler Config object.
+func NewConfig(
+	schedulerCache schedulercache.Cache,
+	ecache *core.EquivalenceCache,
+	nodeLister algorithm.NodeLister,
+	algorithm algorithm.ScheduleAlgorithm,
+	binder Binder,
+	podConditionUpdater PodConditionUpdater,
+	podPreemptor PodPreemptor,
+	nextPod func() *v1.Pod,
+	waitForCacheSync func() bool,
+	error func(*v1.Pod, error),
+	recorder record.EventRecorder,
+	stopEverything chan struct{},
+	volumeBinder *volumebinder.VolumeBinder,
+) *Config {
+	return &Config{
+		schedulerCache:      schedulerCache,
+		ecache:              ecache,
+		nodeLister:          nodeLister,
+		algorithm:           algorithm,
+		binder:              binder,
+		podConditionUpdater: podConditionUpdater,
+		podPreemptor:        podPreemptor,
+		waitForCacheSync:    waitForCacheSync,
+		nextPod:             nextPod,
+		error:               error,
+		Recorder:            recorder,
+		StopEverything:      stopEverything,
+		volumeBinder:        volumeBinder,
+	}
+}
+
+// Algorithm returns the ScheduleAlgorithm object of the config.
+func (c *Config) Algorithm() algorithm.ScheduleAlgorithm {
```

Comment:
Providing getters for "Algorithm" and "SchedulerCache" and not provding getters for "StopEverything" and "Recorder" and leaving the latters publicly accessible make the code inconsistent. Either remove the getters and make the other two publicly accessible or provide getters for all of them.

Reply:
Agreed. I will add getters for `StopEverything` and `Recorder`